Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs

```diff
@@ -0,0 +1,236 @@
+                butcherable.BeingButchered = false;
+        }
+
+        private void OnSpiknigFinished(EntityUid uid, KitchenSpikeComponent component, SpikingFinishedEvent args)
+        {
+            if (EntityManager.TryGetComponent<SharedButcherableComponent>(args.VictimUid, out var butcherable))
+                butcherable.BeingButchered = false;
+
+            if (Spikeable(uid, args.UserUid, args.VictimUid, component, butcherable))
+            {
+                Spike(uid, args.UserUid, args.VictimUid, component);
+            }
+        }
+
+        private void OnDragDrop(EntityUid uid, KitchenSpikeComponent component, DragDropEvent args)
+        {
+            // TODO: Server-side DragDropEvent should be Handleable and use UIDs
+            // if(args.Handled)
+            //      return;
+            //
+            // (!EntityManager.TryGetComponent<SharedButcherableComponent>(args.DraggedUid, out var butcherable)
+
+            if (!EntityManager.TryGetComponent<SharedButcherableComponent>(args.Dragged.Uid, out var butcherable)
+                || butcherable.BeingButchered)
+                return;
+
+            if (!Spikeable(uid, args.User.Uid, args.Dragged.Uid, component))
+                return;
+
+            // if (TrySpike(uid, args.DraggedUid, args.UserUid))
+            //      args.Handled = true;
+
+            TrySpike(uid, args.User.Uid, args.Dragged.Uid, component, butcherable);
+        }
+
+        private void OnInteractUsing(EntityUid uid, KitchenSpikeComponent component, InteractUsingEvent args)
+        {
+            if (args.Handled)
+                return;
+            
+            if (TryGetPiece(uid, args.UserUid, args.UsedUid))
+                args.Handled = true;
+        }
+
+        private void Spike(EntityUid uid, EntityUid userUid, EntityUid victimUid,
+            KitchenSpikeComponent? component = null, SharedButcherableComponent? butcherable = null)
+        {
+            if (!Resolve(uid, ref component) || !Resolve(victimUid, ref butcherable))
+                return;
+
+            component.MeatPrototype = butcherable.MeatPrototype;
+            component.MeatParts = butcherable.Pieces;
+
+            // This feels not okay, but entity is getting deleted on "Spike", for now...
+            component.MeatSource1p = Loc.GetString("sys-kitchen-spike-remove-meat", ("victim", butcherable.Owner));
+            component.MeatSource0 = Loc.GetString("sys-kitchen-spike-remove-meat-last", ("victim", butcherable.Owner));
+            component.MeatName = Loc.GetString("sys-kitchen-spike-meat-name", ("victim", butcherable.Owner));
+
+            UpdateAppearance(uid, null, component);
+
+            var user = EntityManager.GetEntity(uid);
+            _popupSystem.PopupEntity(Loc.GetString("sys-kitchen-spike-kill", ("user", user), ("victim", butcherable.Owner)), uid, Filter.Pvs(userUid));
```

Comment:
Don't use GetEntity, I think there's a way to prettyprint it now.

Reply:
Yea, but I can't use strings with Loc, cuz all the functions there work with IEntity

![image](https://user-images.githubusercontent.com/36813380/144999371-733ce24b-b595-4902-94b6-2bfe2627cbd0.png)

Reply:
nvm, gonna wait for IEntity removal merge to fix this